File: pizza.py
import sys
from wit import Wit

access_token = 'WUBZ5SLRMTIXGXEO74MFTAKIFSB4JWYW'

def first_entity_value(entities, entity):
    if entity not in entities:
        return None
    val = entities[entity][0]['value']
    if not val:
        return None
    return val['value'] if isinstance(val, dict) else val

def send(request, response):
    print(response['text'])

def set_pizza(request):
    context = request['context']
    entities = request['entities']

    pizza_type = first_entity_value(entities,'type')
    pizza_size = first_entity_value(entities,'size')

    if pizza_size and pizza_type:
        context['pizzaType'] = pizza_type
        context['pizzaSize'] = pizza_size
    elif pizza_type and not pizza_size:
        context['pizzaType'] = pizza_type
    elif pizza_size and not pizza_type:
        context['pizzaSize'] = pizza_size
    else:
        context['missingInfo'] = True
    return context

actions = {
    'send': send,
    'setPizza': set_pizza,
}

client = Wit(access_token=access_token, actions=actions)
client.interactive()

File: pizza.js
'use strict';

let Wit = null;
let interactive = null;

Wit = require('node-wit').Wit;
interactive = require('node-wit').interactive;

const accessToken = 'WUBZ5SLRMTIXGXEO74MFTAKIFSB4JWYW';  // WIT access_token

const firstEntityValue = (entities, entity) => {
  const val = entities && entities[entity] &&
    Array.isArray(entities[entity]) &&
    entities[entity].length > 0 &&
    entities[entity][0].value
  ;
  if (!val) {
    return null;
  }
  return typeof val === 'object' ? val.value : val;
};

const actions = {
  send(request, response) {
    const {sessionId, context, entities} = request;
    const {text, quickreplies} = response;
    console.log(JSON.stringify(request));
    console.log('sending...', JSON.stringify(response));
  },
  setPizza({context, entities}) {
    var pizza_type = firstEntityValue(entities, 'type');
    var pizza_size = firstEntityValue(entities, 'size');
    if(pizza_size && pizza_type)
    {
      context.pizzaSize = pizza_size;
      context.pizzaType = pizza_type;
    }
    else if(pizza_type && !pizza_size)
    {
      context.pizzaType = pizza_type;
    }
    else if(pizza_size && !pizza_type)
    {
      context.pizzaSize = pizza_size;
    }
    else
    {
      context.missingInfo = true;
    }

    return context;
  },
};

const client = new Wit({accessToken, actions});
interactive(client);
